### PR TITLE
DIOS-7454  Fixed Safari playback issue for duplicated parameter sets (happens when we re-packetize a video frame)

### PR DESCRIPTION
**Pull request by harryz2000**

### CODE REVIEW

File: src/rtmp/rtmppacketizer.cpp [hunk 1]
```diff
@@ -331,13 +326,47 @@ std::unique_ptr<VideoFrame> RTMPH26xPacketizer<DescClass, SPSClass, PPSClass, co
...
+			auto naluHeader = get2(nal, 0);
+			BYTE nalUnitType = (naluHeader >> 9) & 0b111111;
+			
+			switch (nalUnitType)
 			{
-				auto naluHeader = get2(nal, 0);
-				BYTE nalUnitType = (naluHeader >> 9) & 0b111111;
-				if ((nalUnitType <= HEVC_RTP_NALU_Type::RASL_R) || 
-					(nalUnitType >= HEVC_RTP_NALU_Type::BLA_W_LP && nalUnitType <= HEVC_RTP_NALU_Type::CRA_NUT))
+			case HEVC_RTP_NALU_Type::TRAIL_N:
```
Comment (harryz2000) [inline src/rtmp/rtmppacketizer.cpp:353]: A note here. I think we would prefer to use [Pascal case]( https://khalilstemmler.com/blogs/camel-case-snake-case-pascal-case/#Pascal-case) for constants and enums. I didn't change as these are existing code. Happy to change if it is preferred.

File: src/h264/h264.h [hunk 2]
```diff
@@ -41,6 +41,41 @@ static constexpr uint8_t  Log2MaxMvLengthVerticalMin = 0;
 static constexpr uint8_t  Log2MaxMvLengthVerticalMax = 16;
 static constexpr uint8_t  MaxDpbFrames = 16; // Lifted from FFMPEG
 
+enum AvcNaluType {
```
Comment (murillo128) [inline src/h264/h264.h:44]:
```suggestion
enum class AvcNaluType {
```

File: src/h264/h264.h [hunk 3]
```diff
@@ -41,6 +41,41 @@ static constexpr uint8_t  Log2MaxMvLengthVerticalMin = 0;
...
+	AvcNaluTypeUnspecified		= 0,
```
Comment (murillo128) [inline src/h264/h264.h:45]: remove `AvcNalu` as we are making the `enum` and `enum class` we will access it by `AvcNaluType::Unspecified`

File: src/rtmp/rtmppacketizer.cpp (hunk 1 above)
Comment (murillo128) [reply src/rtmp/rtmppacketizer.cpp:353]: lets change at least the enum name in a later PR

File: src/rtmp/rtmppacketizer.cpp [hunk 4]
```diff
@@ -331,13 +326,47 @@ std::unique_ptr<VideoFrame> RTMPH26xPacketizer<DescClass, SPSClass, PPSClass, co
 						parsingFrameType = false;
 					}
 				}
+				break;
+			case AvcNaluTypeSps:
+				// Skip if we already have sps. Safari may complain about duplicated parameter sets.
```
Comment (murillo128) [inline src/rtmp/rtmppacketizer.cpp:331]:
```suggestion
				// Skip if we already have sps inserted from the AVCDescriptor
				// Safari may complain about duplicated parameter sets.
```

**Review by murillo128 — APPROVED** [review]
lgtm

**Pull request merged**